File: src/main/java/com/slipp/board/api/controller/board/BoardController.java
package com.slipp.board.api.controller.board;

import com.slipp.board.api.service.board.BoardService;
import com.slipp.board.model.Post;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/board")
public class BoardController {

    private final BoardService boardService;

    @Autowired
    public BoardController(BoardService boardService) {
        this.boardService = boardService;
    }

    @RequestMapping(value = "/posts", method = RequestMethod.GET)
    public ResponseEntity<List<Post>> getPosts() {
        List<Post> posts = boardService.getPosts();
        return new ResponseEntity<>(posts, HttpStatus.OK);
    }

    @RequestMapping(value = "/post", method = RequestMethod.POST)
    public ResponseEntity create(@RequestBody Post post) {
        boardService.create(post);
        return new ResponseEntity(HttpStatus.OK);
    }

    @RequestMapping(value = "/post", method = RequestMethod.PUT)
    public ResponseEntity modify(@RequestBody Post post) {
        boardService.modify(post);
        return new ResponseEntity(HttpStatus.OK);
    }

    @RequestMapping(value = "/post/{id}", method = RequestMethod.DELETE)
    public ResponseEntity remove(@PathVariable long id) {
        boardService.remove(id);
        return new ResponseEntity(HttpStatus.OK);
    }
}

File: server/src/main/kotlin/net/slipp/controller/BoardController.kt
package net.slipp.controller

import net.slipp.repository.domain.Board
import net.slipp.service.BoardService
import org.springframework.data.domain.Page
import org.springframework.data.domain.PageRequest
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*
import org.springframework.web.servlet.support.ServletUriComponentsBuilder

/**
 * @author galcyurio
 */
@RestController
@RequestMapping("/boards")
class BoardController(
    private val boardService: BoardService
) {

    @GetMapping
    fun boards(@RequestParam("page") page: Int? = null,
               @RequestParam("size") size: Int? = null): Page<Board> {
        val pageRequest = PageRequest.of(page ?: 0, size ?: 10)
        return boardService.findAll(pageRequest)
    }

    @GetMapping("/{id}")
    fun board(@PathVariable("id") id: Long): ResponseEntity<Board> {
        val board = boardService.findById(id)
        return when (board.isPresent) {
            true -> ResponseEntity.ok(board.get())
            false -> ResponseEntity.notFound().build()
        }
    }

    @PostMapping
    fun save(@RequestBody board: Board): ResponseEntity<Board> {
        val savedBoard = boardService.save(board)
        val location = ServletUriComponentsBuilder
            .fromCurrentRequest().path("/{id}")
            .buildAndExpand(savedBoard.id).toUri()
        return ResponseEntity.created(location).build()
    }

    @PutMapping("/{id}")
    fun update(@PathVariable("id") id: Long, @RequestBody board: Board): ResponseEntity<Board> {
        val updatedBoard = boardService.save(board)
        return ResponseEntity.ok(updatedBoard)
    }

    @DeleteMapping("/{id}")
    fun deleteById(@PathVariable("id") id: Long): ResponseEntity<Any> {
        boardService.deleteById(id)
        return ResponseEntity.ok().build()
    }
}